src/components/MenuBar/MenuBar.tsx

```diff
@@ -0,0 +1,141 @@
+import AppBar from '@material-ui/core/AppBar';
+import Button from '@material-ui/core/Button';
+import IconButton from '@material-ui/core/IconButton';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import MenuIcon from '@material-ui/icons/Menu';
+import type { FunctionComponent, HTMLAttributes } from 'react';
+import React, { Fragment, useContext } from 'react';
+import { Link } from 'react-router-dom';
+
+import styles from '@/components/MenuBar/MenuBar.scss';
+import { friendsConst, loginConst, nameApp, registrationConst, wishConst } from '@/constants';
+import AuthContext from '@/context/AuthContex';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      flexGrow: 1,
+    },
+    menuButton: {
+      marginRight: theme.spacing(2),
+    },
+    title: {
+      flexGrow: 1,
+    },
+  })
+);
+
+interface PropsMenu extends HTMLAttributes<HTMLElement> {
+  children: React.ReactNode;
+}
+
+const MenuBar: FunctionComponent<PropsMenu> = ({ children }) => {
+  const classes = useStyles();
+  const { token, id, logout } = useContext(AuthContext);
+  const profileHref = id ? `/@${id}` : null;
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+
+  const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
```

Comment:
Во всём проекте убери глобальное обращение к реакту, на импорты и раскидай интерфейсы в import type. Так будет проще читать код и все зависимости сразу видны сверху компонента
```suggestion
  const handleMenu = (event: MouseEvent<HTMLElement>) => {
```

Reply:
fix

Reply:
error: Type 'MouseEvent' is not generic.